lib.nix

```diff
@@ -25,19 +27,33 @@ let inherit (pkgs) lib;
         exit 1
       }
     '';
-in rec {
-  src = pkgs.haskell-nix.haskellLib.cleanGit {
+
+  repoSource = pkgs.haskell-nix.haskellLib.cleanGit {
     name = "nix-thunk";
     src = ./.;
   };
 
+  haskellPackageSource = lib.fileset.toSource {
+    fileset = lib.fileset.intersection
+      (lib.fileset.fromSource repoSource)
+      (lib.fileset.unions [
+        ./LICENSE
+        ./CHANGELOG.md
+        ./README.md
+        ./nix-thunk.cabal
+        (lib.fileset.fileFilter (file: file.hasExt "hs") ./src)
+        (lib.fileset.fileFilter (file: file.hasExt "hs") ./src-bin)
+      ]);
+    root = ./.;
+  };
+
```

Comment:
This looks like it will cause surprising and hard-to-debug failures if someone adds more files in the future that cabal depends on, e.g. a `test` directory.  Are we really getting much out of this?

Reply:
Oh good point, we can do better.

(I did this because I was picking up `.hs` files in `dist-newstyle`, but that was before the `lib.fileset.intersection` call. Now that's no longer a problem.)